Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/SpherePointer.cs

```diff
@@ -257,14 +268,81 @@ public bool TryUpdateQueryBufferForLayerMask(LayerMask layerMask, Vector3 pointe
 
                 for (int i = 0; i < numColliders; i++)
                 {
-                    grabbable = queryBuffer[i].GetComponent<NearInteractionGrabbable>();
+                    Collider collider = queryBuffer[i];
+                    grabbable = collider.GetComponent<NearInteractionGrabbable>();
+                    // Additional check: is grabbable in the camera frustrum
                     if (grabbable != null)
+                    {
+                        if (!isInFOV(collider))
+                        {
+                            grabbable = null;
+                        }
+                    }
+                    if (grabbable != null)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+
+            /// <summary>
+            /// Returns true if a collider's bounds is within the camera FOV
+            /// </summary>
+            /// <param name="myCollider">The collider to test</param>
+            /// <returns></returns>
+            private bool isInFOV(Collider myCollider)
+            {
+                Bounds b = myCollider.bounds;
+                var corners = new List<Vector3>();
```

Comment:
If TryUpdateQueryBufferForLayerMask is intended to be called once per frame per the documentation, this would alloc memory each frame right?

Reply:
Good catch, I need to fix that.